Sources/Turf/Spline.swift

```diff
@@ -0,0 +1,147 @@
+    }
+    
+    var coordinate: CLLocationCoordinate2D {
+        return CLLocationCoordinate2D(latitude: y, longitude: x)
+    }
+}
+
+struct Spline {
+    private let points: [SplinePoint]
+    private let duration: Int
+    private let sharpness: Double
+    private let stepLength: Double
+    private let length: Int
+    private let delay: Int = 0
+    private var centers = [SplinePoint]()
+    private var controls = [(SplinePoint, SplinePoint)]()
+    private var steps = [Int]()
+    
+    init?(points: [SplinePoint], duration: Int, sharpness: Double, stepLength: Double = 60) {
+        guard points.count >= 2 else { return nil }
+        self.points = points
+        self.duration = duration
+        self.sharpness = sharpness
+        self.stepLength = stepLength
+        
+        length = points.count
+        
+        for index in stride(from: 0, to: points.count - 1, by: 1) {
+            let point = points[index]
+            let nextPoint = points[index + 1]
+            let center = SplinePoint(x: (point.x + nextPoint.x) / 2, y: (point.y + nextPoint.y) / 2, z: (point.z + nextPoint.z) / 2)
+            centers.append(center)
+        }
+        
+        controls.append((points[0], points[0]))
+        
+        for index in stride(from: 0, to: centers.count - 1, by: 1) {
+            let center = centers[index]
+            let nextCenter = centers[index + 1]
+            let nextPoint = points[index + 1]
+            let dx = nextPoint.x - (center.x + nextCenter.x) / 2
+            let dy = nextPoint.y - (center.y + nextCenter.y) / 2
+            let dz = nextPoint.z - (center.z + nextCenter.z) / 2
+            let control1 = SplinePoint(x: (1 - sharpness) * nextPoint.x + sharpness * (center.x + dx),
+                                       y: (1 - sharpness) * nextPoint.y + sharpness * (center.y + dy),
+                                       z: (1 - sharpness) * nextPoint.z + sharpness * (center.z + dz))
+            let control2 = SplinePoint(x: (1 - sharpness) * nextPoint.x + sharpness * (nextCenter.x + dx),
+                                       y: (1 - sharpness) * nextPoint.y + sharpness * (nextCenter.y + dy),
+                                       z: (1 - sharpness) * nextPoint.z + sharpness * (nextCenter.z + dz))
+            controls.append((control1, control2))
+        }
+        let lastPoint = points.last!
+        controls.append((lastPoint, lastPoint))
+        
+        var lastStep = pos(time: 0)
+        steps.append(0)
+        for t in stride(from: 0, to: duration, by: 10) {
+            let step = pos(time: t)
+            let dist = sqrt((step.x - lastStep.x) * (step.x - lastStep.x) +
+                (step.y - lastStep.y) * (step.y - lastStep.y) +
+                (step.z - lastStep.z) * (step.z - lastStep.z))
+            if dist > stepLength {
+                steps.append(t)
+                lastStep = step
+            }
+        }
+    }
+    
+    func pos(time: Int) -> SplinePoint {
```

Comment:
Nit: rename the method `position(at:)`:

```suggestion
    func position(at time: Int) -> SplinePoint {
```

Reply:
Done